### ISSUE TITLE
"host" config should take precedence over the "socket" config if present

### ISSUE THREAD

**Issue by jjatinggoyal** [trilogy-libraries/trilogy#216]
For initializing db connections, if both `socket` and `host` configs are passed, `trilogy` gives precedence to the `socket` config which results in error messages like below:

```
Trilogy::SyscallError::ENOENT: No such file or directory - trilogy_connect - unable to connect to /tmp/mysql.sock (Trilogy::SyscallError::ENOENT)
```
These can be frustrating to debug since the same config works fine with the `mysql2` adapter but throws an error with `trilogy`.

So if both `socket` and `host` are passed, precedence should be given to the `host` config, and this maintains parity with `mysql2` as well.

**Comment by bensheldon** [comment]
@jjatinggoyal thanks for opening the issue. Other than parity, can you explain why you think `host` should take precedence over `socket`?

**Comment by matthewd** [comment]
We could/should probably just directly refuse to even try to connect when given both? 🤔

**Comment by brianmario** [comment]
> We could/should probably just directly refuse to even try to connect when given both?

This is what I'd personally prefer.

**Comment by jjatinggoyal** [comment]
@bensheldon I stumbled upon this error through a new rails application I created with trilogy. The default `database.yml` in rails looks something like this if you are running MySQL locally.

<img width="393" alt="Screenshot 2024-12-18 at 12 34 00 AM" src="https://github.com/user-attachments/assets/5f7c5101-cb1c-4944-b118-251ebd96e700" />

This works perfectly in the development setup, but when I finally decide to deploy the app to prod, I just add the `host` config as I usually do when using `mysql2`. But I surprisingly ran into the aforementioned error.

So long story short, the confusion is mainly due to the default rails config, my familiarity with `mysql2` and the assumption that `trilogy` is kinda a drop-in replacement for `mysql2`.

**Comment by jjatinggoyal** [comment]
Submitted a [PR](https://github.com/rails/rails/pull/54025) in rails to change the default config for trilogy. Closing this issue.
